### PR TITLE
pci workflow µapp

### PR DESCRIPTION
**Pull request by frenautvh**
| Question         | Answer
| ---------------- | ---
| Branch?          | master
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | DTCORE-2362
| License          | BSD 3-Clause

## Description

pci-workflow µapp

### CODE REVIEW

**Review by anooparveti — CHANGES_REQUESTED** [review]
Can you resolve the conflicts?

**Comment by qpavy** [comment]
You have some issues on the build

**Comment by chipp972** [comment]
You can fix some reliability issues

**Review by qpavy — APPROVED** [review]

File: packages/manager/apps/pci-workflow/src/api/hooks/useInstances.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  filters: Filter[] = [],
+) => {
+  const { data: instances, error, isPending } = useAllInstances(projectId);
+  return useMemo(
+    () => ({
+      isPending,
+      error,
+      data: paginateResults<TInstance>(
+        sortResults(
+          applyFilters(((instances as unknown) as TInstance[]) || [], filters),
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-workflow/src/api/hooks/useInstances.ts:93]: The cast should appear before

File: packages/manager/apps/pci-workflow/src/hooks/usePageTracking.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        projectId: project.project_id,
+        isDiscoveryProject: project.planCode === DISCOVERY_PLANCODE,
+      });
+    }
+  }, [project]);
+
+  useEffect(() => {
+    const pageId = location.pathname.split('/').pop();
+    const pageKey = pageId === 'workflow' ? '' : `::${pageId}`;
+    trackPage({
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-workflow/src/hooks/usePageTracking.ts:28]: Should this be moved to useOvhTracking ?

File: packages/manager/apps/pci-workflow/src/api/hooks/useInstances.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import { useQueries, UseQueryResult } from '@tanstack/react-query';
+import { useTranslatedMicroRegions } from '@ovh-ux/manager-react-components';
+import { sortResults, TInstanceOptions } from '@/api/data/instance';
+import { paginateResults } from '@/helpers';
+import { TWorkflowInstance } from '@/types';
+
+export const getStatusGroup = (status: string) => {
+  if (
+    [
+      'BUILDING',
```
Comment (tristanwagner) [inline packages/manager/apps/pci-workflow/src/api/hooks/useInstances.ts:13]: might want to declare an enum for that

Comment (frenautvh) [reply packages/manager/apps/pci-workflow/src/api/hooks/useInstances.ts:13]: typescript string enums are very verbose, also we have to check the status at runtime and not at compilation time so i'm not sure about it

could you provide an example of what you consider a better way of handling this case ? thanks

File: packages/manager/apps/pci-workflow/src/api/hooks/useInstances.ts (hunk 1 above)
Comment (frenautvh) [reply packages/manager/apps/pci-workflow/src/api/hooks/useInstances.ts]: removed the cast as it was not needed anymore, ty

File: packages/manager/apps/pci-workflow/src/hooks/usePageTracking.ts (hunk 2 above)
Comment (frenautvh) [reply packages/manager/apps/pci-workflow/src/hooks/usePageTracking.ts:28]: for the migration our goal is to have the same tracking behavior as the angularjs section, we are not implementing new tracking behaviors

**Review by anooparveti — APPROVED** [review]

**Pull request merged**